apps/easypid/src/features/didcomm/DidCommNotificationScreen.tsx

```diff
@@ -31,8 +35,43 @@ export function DidCommNotificationScreen() {
     type: 'credentialExchange' | 'proofExchange'
   }>()
 
-  const onCancel = () => pushToWallet('back')
-  const onComplete = () => pushToWallet('replace')
+  const handleNavigation = (type: 'replace' | 'back') => {
+    // When starting from the inbox, we want to go back to the inbox on finish
+    if (params.navigationType === 'inbox') {
+      router.back()
+    } else {
+      pushToWallet(type)
+    }
+  }
+
+  const onCancel = () => {
+    if (notification) {
+      if (notification.type === 'credentialExchange' || params.credentialExchangeId) {
+        void agent.modules.credentials.deleteById(notification.id ?? params.credentialExchangeId)
+      } else if (notification.type === 'proofExchange' || params.proofExchangeId) {
+        void agent.modules.proofs.deleteById(notification.id ?? params.proofExchangeId)
+      }
```

Comment:
this cod  will only execute if there is a notification. In which case is notification defined, but notificadtion.id is undefined, and it will fallback to params?

Also, can't we populate `notification` based on the params? that way we can handle them in the same way here